src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -161,9 +170,9 @@ def run(
         """
         try:
             self._start_time = datetime.utcnow()
-            with _change_working_dir(self._working_dir):
+            with (_change_working_dir(self._working_dir)):
                 # create executor proxy instance according to the flow program language
-                self._executor_proxy = ProxyFactory().create_executor_proxy(
+                self._executor_proxy = executor_proxy or ProxyFactory().create_executor_proxy(
```

Comment:
is it possible that we avoid `or` here? it may bring convenience for current PR but can make it hard in further maintain.

We can always initialize executor proxy out of batch engine according to current design.

On the other hand, is `self._executor_proxy` still necessary? I don't think it's a good idea to set a private member in a `run` function.

cc @thy09  @PeiwenGaoMS